raiden/tests/unit/transfer/mediated_transfer/test_mediator.py

```diff
@@ -0,0 +1,1300 @@
+    for payer, payee in zip(hops[:-1], hops[1:]):
+        assert next_expiration > 0
+
+        # for simplicity factory doesn't have a facility to each transfer
+        # state, so even the secret might be known the state is kept at pending
+        pair = make_transfer_pair(
+            payer,
+            payee,
+            target,
+            amount,
+            next_expiration,
+            secret=secret,
+            reveal_timeout=reveal_timeout,
+        )
+        transfers_pair.append(pair)
+
+        # assumes that the node sending the refund will follow the protocol and
+        # decrement the expiration for it's lock
+        next_expiration = pair.payee_transfer.expiration - reveal_timeout
+
+    return transfers_pair
+
+
+def make_mediator_state(from_transfer,
+                        from_route,
+                        routes,
+                        our_address=factories.ADDR):
+
+    state_change = make_init_statechange(
+        from_transfer,
+        from_route,
+        routes,
+        our_address=our_address,
+    )
+
+    initial_state = None
+    iteration = mediator.state_transition(initial_state, state_change)
+
+    return iteration.new_state
+
+
+def test_is_lock_valid():
+    """ A hash time lock is valid up to the expiraiton block. """
+    amount = 10
+    expiration = 10
+    transfer = factories.make_transfer(amount, factories.HOP1, expiration)
+
+    assert mediator.is_lock_valid(transfer, 5) is True
+    assert mediator.is_lock_valid(transfer, 10) is True, 'lock is expired at the next block'
+    assert mediator.is_lock_valid(transfer, 11) is False
+
+
+def test_is_safe_to_wait():
+    """ It's safe to wait for a secret while there are more than reveal timeout
+    blocks until the lock expiration.
+    """
+    amount = 10
+    expiration = 40
+    transfer = factories.make_transfer(amount, factories.HOP1, expiration)
+
+    # expiration is in 30 blocks, 19 blocks safe for waiting
+    block_number = 10
+    reveal_timeout = 10
+    assert mediator.is_safe_to_wait(transfer, reveal_timeout, block_number) is True
+
+    # expiration is in 30 blocks, 09 blocks safe for waiting
+    block_number = 20
+    reveal_timeout = 10
+    assert mediator.is_safe_to_wait(transfer, reveal_timeout, block_number) is True
+
+    # expiration is in 30 blocks, 1 block safe for waiting
+    block_number = 29
+    reveal_timeout = 10
+    assert mediator.is_safe_to_wait(transfer, reveal_timeout, block_number) is True
+
+    # at the block 30 it's not safe to wait anymore
+    block_number = 30
+    reveal_timeout = 10
+    assert mediator.is_safe_to_wait(transfer, reveal_timeout, block_number) is False
+
+    block_number = 40
+    reveal_timeout = 10
+    assert mediator.is_safe_to_wait(transfer, reveal_timeout, block_number) is False
+
+    block_number = 50
+    reveal_timeout = 10
+    assert mediator.is_safe_to_wait(transfer, reveal_timeout, block_number) is False
+
+
+def test_is_channel_close_needed_unpaid():
```

Comment:
so...my 2 cents here: since `test_<function_name>_<condition>` tends to become really long in this test suite, and sometimes hard to parse, it may be helpful to use a double `__` underscore between function name and condition: `test_<function_name>__<condition>`. Would that make it easier to take the name, @LefterisJP ?